legacy-script.sh fixes - junit output and add missing record_command

brianpursley opened the pull request:
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
I was recently reviewing a PR related to the tests in legacy-script.sh and noticed a few problems in the existing code.  This PR fixes those problems.

1.  Added `record_command` to the following tests in legacy-script.sh:
       * run_recursive_resources_tests
       * run_deprecated_api_tests

2.  Added code to set `KUBE_JUNIT_REPORT_DIR` (copied from `hack/make-rules/test.sh`) to produce junit output in the artifacts directory, so that the individual test results will show up in the build pipeline output's junit section.

3.  Fixed some typos in README.md pertaining to how to run individual integration tests.

#### Which issue(s) this PR fixes:

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

brianpursley commented:
/retest

brianpursley commented:
/assign soltysh

oomichi commented on test/cmd/legacy-script.sh:
```diff
@@ -106,6 +106,12 @@ daemonsets="daemonsets"
 job="jobs"
 
+# Create a junit-style XML test report in this directory if set.
+KUBE_JUNIT_REPORT_DIR=${KUBE_JUNIT_REPORT_DIR:-}
```
This line sets KUBE_JUNIT_REPORT_DIR from the same variable name with default empty string.
I could not find the reason why this line is necessary.

brianpursley replied:
You know, I think you're right.

I had copied that code block from here: https://github.com/kubernetes/kubernetes/blob/69c3e1934c0c2b01c53e765cb1f5ef8b4b95dfd6/hack/make-rules/test.sh#L80-L85

But taking a closer look at it, I agree.  It doesn't appear to do anything.

oomichi replied:
Ah, OK. I missed the above code.
Thanks for updating.

oomichi reviewed (commented):
Most part seems good for me, just one question.

/cc @oomichi

brianpursley committed:
Change legacy-script.sh to produce junit output in location set by the
ARTIFACTS environment variables if KUBE_JUNIT_REPORT_DIR is not set.

Add missing record_command to a couple of tests in legacy-script.sh.

Fix some typos in /test/cmd/README.md

brianpursley commented:
/retest

oomichi commented:
/lgtm

oomichi commented:
/triage accepted
